modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java

```diff
@@ -539,6 +570,14 @@ else if (Objects.equals(BinaryExpression.Operation.OR, operation)) {
 				Predicate.withParentheses((Predicate)left),
 				Predicate.withParentheses((Predicate)right));
 		}
+		else if (Objects.equals(left, "keywords") &&
```

Comment:
@LuismiBarcos, @ccorreagg I know this `if` is ugly. But, since we will be implementing categories next, I'd rather wait till that point to see how it looks like and come up with a better way to do this. Maybe some kind of `externalFieldPredicateProvider` or something like that similar to what @LuismiBarcos did for relationships.

Reply:
Yeah, I would like to see how the code will be with both parameters, but I really feel that if is ugly. Maybe we should a way to generalize the way we read from the table based on the field